# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1208696687 14400
#      Sun Apr 20 09:04:47 2008 -0400
# Branch release-3-0-x
# Node ID 61f955d848c30f2ee768209fbfb798759ccd41e0
# Parent  d68fc0ca4ecec87c0d08de759221843a2616a6cb
octave_stream::read: allow single data type specification but return double

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2008-04-20  John W. Eaton  <jwe@octave.org>
+
+	* oct-stream.cc (octave_stream::read): Allow single data type
+	specification but return double.
+
 2008-04-18  John W. Eaton  <jwe@octave.org>
 
 	* pr-output.cc (Fdisp): If nargout > 0, produce an sq-string
 	unless arg is a dq-string.
 
 2008-04-14  Jaroslav Hajek <highegg@gmail.com>
 
 	* oct-stream.cc (octave_scan_1): Ensure digit following X is hex
diff --git a/src/oct-stream.cc b/src/oct-stream.cc
--- a/src/oct-stream.cc
+++ b/src/oct-stream.cc
@@ -3247,16 +3247,20 @@ octave_stream::read (const Array<double>
       FILL_TABLE_ROW (oct_data_conv::dt_int8, int8NDArray);
       FILL_TABLE_ROW (oct_data_conv::dt_uint8, uint8NDArray);
       FILL_TABLE_ROW (oct_data_conv::dt_int16, int16NDArray);
       FILL_TABLE_ROW (oct_data_conv::dt_uint16, uint16NDArray);
       FILL_TABLE_ROW (oct_data_conv::dt_int32, int32NDArray);
       FILL_TABLE_ROW (oct_data_conv::dt_uint32, uint32NDArray);
       FILL_TABLE_ROW (oct_data_conv::dt_int64, int64NDArray);
       FILL_TABLE_ROW (oct_data_conv::dt_uint64, uint64NDArray);
+      // FIXME -- the following line allows things like int8=>single
+      // to work, but they will actually return a double value.  We
+      // need a floatNDArray for this to work properly.
+      FILL_TABLE_ROW (oct_data_conv::dt_single, NDArray);
       FILL_TABLE_ROW (oct_data_conv::dt_double, NDArray);
       FILL_TABLE_ROW (oct_data_conv::dt_char, charNDArray);
       FILL_TABLE_ROW (oct_data_conv::dt_schar, charNDArray);
       FILL_TABLE_ROW (oct_data_conv::dt_uchar, charNDArray);
       FILL_TABLE_ROW (oct_data_conv::dt_logical, boolNDArray);
 
       initialized = true;
     }
